src/node_jsrtapi.cc

```diff
@@ -364,69 +403,78 @@ napi_status napi_set_property(napi_env e, napi_value o,
   return napi_ok;
 }
 
-napi_status napi_define_property(napi_env e, napi_value o,
-    napi_property_descriptor* p) {
-  napi_value descriptor;
-  CHECK_NAPI(napi_create_object(e, &descriptor));
-
+napi_status napi_define_properties(napi_env e,
+                                   napi_value o,
+                                   int property_count,
+                                   const napi_property_descriptor* properties) {
   napi_propertyname configurableProperty;
   CHECK_NAPI(napi_property_name(e, "configurable", &configurableProperty));
```

Comment:
I'd be in favor of eliminating napi_propertyname too. @gabrielschulhof @mhdawson @sampsongao thoughts?

Convenience I can see- what is the maintainability argument? I don't have strong feelings one way or the other but it does seem like an additional layer of abstraction that could hurt perf in some cases- and it makes the code slightly weirder to read, seeing a mix of JSRT and NAPI calls in a single method implementation- I can see the calls to NAPI if the equivalent JSRT methods don't exist, but in the event that they do, it triggers the OCD part of me 😄